src/mlpack/tests/init_rules_test.cpp

```diff
@@ -197,6 +198,108 @@ BOOST_AUTO_TEST_CASE(GaussianInitTest)
   BOOST_REQUIRE_EQUAL(weights3d.n_slices, slices);
 }
 
+/**
+ * Simple test of the VarianceScalingNormalInit class.
+ * The test checks if the size of initialized weight matrices is correct.
+ * Further it checks the size of initialized weight cubes is correct.
+ * It also checks whether the values initialized make sense by verifying if
+ * values lie within 3 standard deviations of the mean of normal distribution.
+ */
+BOOST_AUTO_TEST_CASE(VarianceScalingNormalInitTest)
+{
+  const size_t rows = 7;
+  const size_t cols = 8;
+  const size_t slices = 2;
+
+  arma::mat weights;
+  arma::cube weights3d;
+
+  VarianceScalingNormalInit<> vsn;
+
+  vsn.Initialize(weights, rows, cols);
+  vsn.Initialize(weights3d, rows, cols, slices);
+
+
+  // Check that size of initialized weights is equal to the input weights size
+
+  BOOST_REQUIRE_EQUAL(weights.n_rows, rows);
+  BOOST_REQUIRE_EQUAL(weights.n_cols, cols);
+
+  BOOST_REQUIRE_EQUAL(weights3d.n_rows, rows);
+  BOOST_REQUIRE_EQUAL(weights3d.n_cols, cols);
+  BOOST_REQUIRE_EQUAL(weights3d.n_slices, slices);
+
+  // Check that the initialized values lie in the required range
+
+  double variance = (1.0 / ((rows + cols) / 2.0));
+  double stddev = sqrt(variance);
+
+  for (size_t i = 0; i < rows; i++)
+    for (size_t j = 0; j < cols; j++)
+    {
+      BOOST_REQUIRE_GE(weights.at(i, j), - 3 * stddev);
+      BOOST_REQUIRE_LE(weights.at(i, j), 3 * stddev);
+    }
+
+  for (size_t k = 0; k < slices; k++)
+    for (size_t i = 0; i < rows; i++)
+      for (size_t j = 0; j < cols; j++)
+      {
+        BOOST_REQUIRE_GE(weights3d.slice(k).at(i, j), - 3 * stddev);
+        BOOST_REQUIRE_LE(weights3d.slice(k).at(i, j), 3 * stddev);
+      }
+}
+
+/**
+ * Simple test of the VarianceScalingUniformInit class.
+ * The test checks if the size of initialized weight matrices is correct.
+ * Further it checks the size of initialized weight cubes is correct.
+ * It also checks whether the values initialized make sense by verifying if
+ * values lie within the limits of the uniform initialization.
+ */
+BOOST_AUTO_TEST_CASE(VarianceScalingUniformInitTest)
+{
+  const size_t rows = 7;
+  const size_t cols = 8;
+  const size_t slices = 2;
+
+  arma::mat weights;
+  arma::cube weights3d;
+
+  VarianceScalingUniformInit<> vsu;
+
+  vsu.Initialize(weights, rows, cols);
+  vsu.Initialize(weights3d, rows, cols, slices);
+
+  // Check that size of initialized weights is equal to the input weights size
+
+  BOOST_REQUIRE_EQUAL(weights.n_rows, rows);
+  BOOST_REQUIRE_EQUAL(weights.n_cols, cols);
+
+  BOOST_REQUIRE_EQUAL(weights3d.n_rows, rows);
+  BOOST_REQUIRE_EQUAL(weights3d.n_cols, cols);
+  BOOST_REQUIRE_EQUAL(weights3d.n_slices, slices);
+
+  // Check that the initialized values lie in the required range
+
+  double limit = sqrt(3.0 / ((rows + cols) / 2.0));
+
+  for (size_t i = 0; i < rows; i++)
```

Comment:
We could use ``arma::all`` to simplify the test here. Let me know what you think.

Reply:
arma:all does make sense. It should reduce all the loops.